src/codegen/codegen_compatibility_visitor.cpp

```diff
@@ -5,12 +5,25 @@
  * Lesser General Public License. See top-level LICENSE file for details.
  *************************************************************************/
 
-#include <algorithm>
-#include <iostream>
-#include <memory>
-#include <string>
-
 #include "codegen/codegen_compatibility_visitor.hpp"
+
+#include "ast/after_block.hpp"
+#include "ast/before_block.hpp"
+#include "ast/constant_block.hpp"
+#include "ast/constructor_block.hpp"
+#include "ast/destructor_block.hpp"
+#include "ast/discrete_block.hpp"
+#include "ast/function_table_block.hpp"
+#include "ast/global.hpp"
+#include "ast/independent_block.hpp"
+#include "ast/match_block.hpp"
+#include "ast/name.hpp"
+#include "ast/partial_block.hpp"
+#include "ast/pointer_var.hpp"
+#include "ast/solve_block.hpp"
+#include "ast/string.hpp"
+#include "ast/terminal_block.hpp"
+#include "ast/verbatim.hpp"
```

Comment:
Thanks @tristan0x ! I will take a detailed look but did you see my comment in https://github.com/BlueBrain/nmodl/issues/298#issuecomment-614939876?

Reply:
I half-read it apparently. I didn't see your proposal...
Indeed this patch doesn't address the issues with pyast.cpp and pyvisitor.cpp but I experience a 10-15% build time improvements on my imac.

Reply:
As this is a significant change, I would mark this PR as `changes requested` until I test this thoroughly . My first though is that we have around >120 AST classes in the NMODL. Having those as a separate header and including individually is bit inconvenient (I will elaborate this later).

Reply:
One improvement I can think of is to generate one additional header for every group of classes under Ast with a maximum depth of 1 or 2 i.e:

* `ast/expressions.hpp`
  * `ast/numbers.hpp`
  * `ast/identifiers.hpp`
  * `ast/blocks.hpp`
* `ast/statements.hpp`

Reply:
this suggested we will incorporate into separate PR.